src/websockets-server.cpp

```diff
@@ -159,16 +159,13 @@ int32_t WebsocketServer::callbackHttp(struct lws *wsi, enum lws_callback_reasons
     lws_callback_on_writable(wsi);
 
   } else if (reason == LWS_CALLBACK_HTTP_WRITEABLE) {
-    std::string content = clientData->httpResponse->getContent() + "\n";
-    
-    unsigned char *contentBuf = new unsigned char[content.length() + 2];
-    strcpy((char *)contentBuf, content.c_str());
-
-    lws_write(wsi, contentBuf, content.length(), LWS_WRITE_HTTP);
+    std::string const CONTENT = clientData->httpResponse->getContent() + "\n";
+    uint32_t const LEN = CONTENT.length();
+    unsigned char *contentBuf = new unsigned char[LEN];
+    memcpy(contentBuf, CONTENT.c_str(), LEN);
+    lws_write(wsi, contentBuf, LEN, LWS_WRITE_HTTP);
     delete[] contentBuf;
```

Comment:
Cf. question above.

Reply:
Why the same comment twice?

Reply:
I was referring to the delete call; when we use pre-allocated memory, we wouldn’t need this

Reply:
It's also very hard to know the size of that buffer, since the files sent by the server might very quite much in size. Any suggestion how to solve it?

Reply:
I would suggest to use a good default value from our experience as pre-allocated memory; only if the data to be sent is larger, we do a dynamic allocation.

Reply:
I wonder if this is really worth it. The HTTP requests are very low frequency, and I would need to pre-allocate for each client that connects, and also remove the allocation if the client later is disconnected. The risk for problems here seems larger than the gain. Perhaps it would make sense if we are foreseeing thousands of clients, but I don't know...

Reply:
If not doing it per-client I would need to mutex that memory up, and that would probably slow things down even more than the dynamic allocation. Or what do you think?

Reply:
I can follow your reasoning. Let’s leave it for now and we monitor it.